### PR TITLE
Usergroups

### PR DESCRIPTION
**Pull request by marksou**
Added User Groups page for Creating a group via the Dashboard.

### CODE REVIEW

File: tyk-docs/content/dashboard-admin-api/sso.md [hunk 1]
```diff
@@ -15,15 +15,16 @@ Our Tyk Identity Broker (TIB) internally also uses this API.
...
+* `OrgID` - with your organisation id.
+* `GroupID` - the group id 
 
 
 | **Property** | **Description**              |
 | ------------ | ---------------------------- |
 | Resource URL | `/admin/sso` |
 | Method       | POST                         |
-| Body         | `{"ForSection":"<scope>", "OrgID": "<org-id>"}`  |
+| Body         | `{"ForSection":"<scope>", "OrgID": "<org-id>", "GroupID": "<org-id>"}`  |
```
Comment (buger) [inline tyk-docs/content/dashboard-admin-api/sso.md]: Typo, should be `<group-id>`

File: tyk-docs/content/dashboard-admin-api/sso.md [hunk 2]
```diff
@@ -34,7 +35,8 @@ admin-auth: 12345
     
 {
     "ForSection": "dashboard",
-    "OrgID": "588b4f0bb275ff0001cc7471"
+    "OrgID": "588b4f0bb275ff0001cc7471",
+    "group_id": ""
```
Comment (buger) [inline tyk-docs/content/dashboard-admin-api/sso.md]: Should be "GroupID"

File: tyk-docs/content/security/dashboard/create-user-groups.md [hunk 3]
```diff
@@ -0,0 +1,57 @@
+---
+date: 2017-03-23T14:59:47Z
+title: Create User Groups
```
Comment (buger) [inline tyk-docs/content/security/dashboard/create-user-groups.md:3]: From where we link this page? Or it just visible in navigation?

**Comment by buger** [comment]
We should add paragraph about user groups to https://tyk.io/docs/security/dashboard/user-roles/

File: tyk-docs/content/security/dashboard/create-user-groups.md (hunk 3 above)
Comment (marksou) [reply tyk-docs/content/security/dashboard/create-user-groups.md:3]: Just visible in navigation at the moment

File: tyk-docs/content/dashboard-admin-api/sso.md [hunk 4]
```diff
@@ -15,15 +15,16 @@ Our Tyk Identity Broker (TIB) internally also uses this API.
...
+* `OrgID` - with your organisation id.
+* `GroupID` - the group id 
 
 
 | **Property** | **Description**              |
 | ------------ | ---------------------------- |
 | Resource URL | `/admin/sso` |
 | Method       | POST                         |
-| Body         | `{"ForSection":"<scope>", "OrgID": "<org-id>"}`  |
+| Body         | `{"ForSection":"<scope>", "OrgID": "<org-id>", "group-id": "<org-id>"}`  |
```
Comment (buger) [inline tyk-docs/content/dashboard-admin-api/sso.md]: You misunderstood a bit, it should be: `"GroupID": "<group-id>"`

File: tyk-docs/content/security/dashboard/user-roles.md [hunk 5]
```diff
@@ -48,5 +49,7 @@ Permissions are enforced **at the Dashboard API level**.
 
 Each of the object categories will also have an effect on the dashboard navigation, however side-effects can occur if pages that make use of multiple APIs to fetch configuration data cross over e.g. policies and API Definition listings.
 
-[1]: /docs/img/dashboard/system-management/admin_account_2.5.png
+From v2.7 you can now assign users to a user group if you are an admin user or have the **User Group** permission assigned to you. See [Create User Groups](https://tyk.io/docs/security/dashboard/create-user-groups/) for more details.
```
Comment (buger) [inline tyk-docs/content/security/dashboard/user-roles.md]: I think you should move it on top, since it is important. And maybe rephrase it to smth like:

```
The Tyk Dashboard allows multi-tenant and granular user access. Users can be assigned specific permissions to ensure that they only have very specific access to the dashboard pages, and to the underlying API.

Starting with Tyk Dashboard 1.7 instead of setting permissions per user, you can now [create a user group](put link here) and assign it to multiple users. An example can be creating the group for users who can only see analytics, or only manage portal developers. 

> This feature is available to all our Cloud and Hybrid users. For On-Premise installations, this feature is available only for customers with “Unlimited” license.

It is important to note that all user roles are defined and enforced at the API level, and the UI is merely reactive.
```

File: tyk-docs/content/security/dashboard/create-user-groups.md (hunk 3 above)
Comment (buger) [reply tyk-docs/content/security/dashboard/create-user-groups.md:3]: Maybe rename title to "User groups"?

File: tyk-docs/content/security/dashboard/user-roles.md [hunk 6]
```diff
@@ -33,7 +37,8 @@ The permissions object, when fully set as an API entry or in MongoDB, looks like
         "policy": "write",
         "portal": "write",
         "system": "write",
-        "users": "write"
+        "users": "write",
+        "group_id": "write"
```
Comment (buger) [inline tyk-docs/content/security/dashboard/user-roles.md]: should be: `"user_groups": "write"`

File: tyk-docs/content/dashboard-admin-api/sso.md [hunk 7]
```diff
@@ -15,15 +15,16 @@ Our Tyk Identity Broker (TIB) internally also uses this API.
 The Dashboard exposes the `/admin/sso` Admin API which allows you to generate a temporary authentication token, valid for 60 seconds. 
 
 You should provide JSON payload with the following data:
-- `ForSection` - scope with possible values of `"dashboard"` or `"portal"`. 
-- `OrgID`      - with your organisation id.
+* `ForSection` - scope with possible values of `"dashboard"` or `"portal"`. 
+* `OrgID` - with your organisation id.
```
Comment (buger) [inline tyk-docs/content/dashboard-admin-api/sso.md:19]: Add `EmailAddress` here too, with the description: "User email"

File: tyk-docs/content/dashboard-admin-api/sso.md [hunk 8]
```diff
@@ -15,15 +15,16 @@ Our Tyk Identity Broker (TIB) internally also uses this API.
...
+* `GroupID` - the group id 
 
 
 | **Property** | **Description**              |
 | ------------ | ---------------------------- |
 | Resource URL | `/admin/sso` |
 | Method       | POST                         |
-| Body         | `{"ForSection":"<scope>", "OrgID": "<org-id>"}`  |
+| Body         | `{"ForSection":"<scope>", "OrgID": "<org-id>", ""GroupID": "<group-id>"}`  |
```
Comment (buger) [inline tyk-docs/content/dashboard-admin-api/sso.md]: Redundant `"` symbol.
